xray/plot/plot.py

```diff
@@ -287,21 +288,40 @@ def _determine_extend(calc_data, vmin, vmax):
 
 def _color_palette(cmap, n_colors):
     import matplotlib.pyplot as plt
-    try:
-        from seaborn.apionly import color_palette
-        pal = color_palette(cmap, n_colors=n_colors)
-    except (TypeError, ImportError, ValueError):
```

Comment:
The _color_palette function was a big source of confusion for be because it served two very different purposes:

1. It translated a list of colors into a palette using seaborn
2. It translated named colormaps (e.g. `jet`) into palettes, including possibly seaborn palettes

I have refactored the function a little to make these two roles more clear.
It turns out that seaborn is not needed for 1. It can be done with ``matplotlib.colors.ListColormap``. I have left the seaborn version of this here for now, but if it does the same thing as matplotlib, it can be removed.